bait-files/src/main/java/com/google/sps/servlets/RequestServlet.java

```diff
@@ -40,24 +49,46 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     Map<String, String[]> parameters = request.getParameterMap();
     Long requestId = null;
 
-    //get user id
+    // get user id
     UserService userService = UserServiceFactory.getUserService();
     String userId = userService.getCurrentUser().getUserId();
 
     String nameInput = parameters.get(NAME)[0];
     String usernameInput = parameters.get(USERNAME)[0];
     String emailInput = parameters.get(EMAIL)[0];
     String addressInput = parameters.get(ADDRESS)[0];
-    String pictureInput = parameters.get(PICTURE)[0];
+    String blobkeyString = getUploadedFileUrl(request, PICTURE);
     String phoneInput = parameters.get(PHONE)[0];
     String notesInput = parameters.get(NOTES)[0];
 
     ObjectifyService.ofy()
         .save()
         .entity(new Request(requestId, userId, nameInput, usernameInput, emailInput, addressInput,
-            pictureInput, phoneInput, notesInput))
+            blobkeyString, phoneInput, notesInput))
         .now();
 
     response.sendRedirect("/submission.html");
   }
+
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
```

Comment:
Maybe add a comment somewhere around here that the form restricts the filetype, so we don't need to check here.